src/util/spreadsheet/misc.py

```diff
@@ -1,32 +1,33 @@
-from .. import json
+from .. import json, pg
 
 
 def iter_commands(cr, like_all=(), like_any=()):
     if not (bool(like_all) ^ bool(like_any)):
         raise ValueError("Please specify `like_all` or `like_any`, not both")
-    cr.execute(
-        """
-        SELECT id,
-               commands
-          FROM spreadsheet_revision
-         WHERE commands LIKE {}(%s::text[])
-        """.format("ALL" if like_all else "ANY"),
-        [list(like_all or like_any)],
-    )
-    for revision_id, data in cr.fetchall():
-        data_loaded = json.loads(data)
-        if "commands" not in data_loaded:
-            continue
-        data_old = json.dumps(data_loaded, sort_keys=True)
+    with pg.named_cursor(cr, itersize=1) as ncr:
```

Comment:
You can either leave the default `itersize`, or optimize the value to something that would work depending on the data. Another alternative is to use `fetchmany` directly.
```suggestion
    with pg.named_cursor(cr) as ncr:
```